analysis/type.go

```diff
@@ -73,7 +73,89 @@ type TokenFilter interface {
 }
 
 type Analyzer interface {
-	Analyze([]byte) TokenStream
+	Type() string
+	// return value of this method depends on the type of analyzer
+	Analyze([]byte) interface{}
+}
+
+const (
+	TokensAnalyzerType     = "token"
+	HookTokensAnalyzerType = "hook_token"
+	VectorAnalyzerType     = "vector"
+	HookVectorAnalyzerType = "hook_vector"
+)
+
+// # Analyzer Analyze() return type
+type TokensAnalyzer struct {
+	Tokens TokenStream
+}
+type HookTokensAnalyzer struct {
+	Tokens TokenStream
+	Err    error
+}
+type VectorAnalyzer []float64
+type HookVectorAnalyzer struct {
+	Vector []float64
+	Err    error
+}
+
+func AnalyzeForTokens(analyzer Analyzer, input []byte) (TokenStream, error) {
```

Comment:
add comment:
// A utility function, helpful for analyzing an input to generate TokenStream ( and error, if any )

Previously, Analyze() method of an analyzed to return TokenStream.
But as per the change in this PR, Analyze() method will now return a value of type interface{} , validating and using which can be done based on analyzer.Type().
For the benefit of  users of old Analyzer interface, this utiity will come handly , to migrate to new Analyzer interface.